### PR TITLE
Settings menu to master, Closes #26

### LINKED ISSUE #26: Fix UI scaling issues on different mobile devices

### PR DESCRIPTION
**Pull request by danwardvs**

### CODE REVIEW

**Comment by alegemaate** [comment]
closes #23 
Pls 2 see comments

File: BallBounceGame/Assets/GameStart/GameScript.cs [hunk 1]
```diff
@@ -40,7 +41,13 @@ public void SetLevelFinish(bool newFinish)
     
     // Start is called before the first frame update
     void Start()
-    {
+    {   
+
+        // Load inverted controls from PlayerPrefs, which is stored in a local
+        // file and is set by the settings menu. 
+        if(PlayerPrefs.GetInt("Control", 1)==0){
```
Comment (alegemaate) [inline BallBounceGame/Assets/GameStart/GameScript.cs:48]: petition to autoformat code

File: BallBounceGame/Assets/SettingsGUI/SettingsScript.cs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        music_button.colors = new_cb;
+        music_button.GetComponentInChildren<Text>().text = new_text;
+
+    }
+
+    public void SFXButtonPressed(){
+
+        sfx_enabled = !sfx_enabled;
+
+        PlayerPrefs.SetInt("Sound", sfx_enabled ? 1 : 0);
```
Comment (alegemaate) [inline BallBounceGame/Assets/SettingsGUI/SettingsScript.cs:71]: Nice use of shorthand

File: BallBounceGame/Assets/SettingsGUI/SettingsScript.cs [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            new_cb.highlightedColor = new Color(0.3f,1f,0.3f);
+            new_text = "YES";
+        }else{
+            new_cb.normalColor = Color.red;
+            new_cb.highlightedColor = new Color(1f,0.3f,0.3f);
+            new_text = "MUTE";
+        }
+
+        sfx_button.colors = new_cb;
+        sfx_button.GetComponentInChildren<Text>().text = new_text;
```
Comment (alegemaate) [inline BallBounceGame/Assets/SettingsGUI/SettingsScript.cs]: Would be ideal to set directly, use shorthand. ex.
```
sfx_button.GetComponentInChildren<Text>().text = sfx_enabled ? "YES" : "NO";
sfx_button.colors.normalColor = sfx_enabled ? Color.green : Color.red;
sfx_button.colors.highlightedColor = sfx_enabled ? new Color(0.3f,1f,0.3f) : new Color(1f,0.3f,0.3f);
```
or, use if else structure but assign directly. Less clutter IMO, but not necessary

File: BallBounceGame/Assets/TitleScreen/TitleScreenScript.cs [hunk 4]
```diff
@@ -19,6 +19,10 @@ public void QuitGame()
     {
         Application.Quit();
     }
+    public void OpenSettings(){
+        SceneManager.LoadScene(2, LoadSceneMode.Single);
+
```
Comment (alegemaate) [inline BallBounceGame/Assets/TitleScreen/TitleScreenScript.cs:24]: Formatting gone rouge

File: BallBounceGame/Assets/SettingsGUI/SettingsScript.cs [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    public void MusicButtonPressed(){
+
+        music_enabled = !music_enabled;
+
+        PlayerPrefs.SetInt("Music", music_enabled ? 1 : 0);
+
+        ColorBlock new_cb = music_button.colors;
+        string new_text;
+
+        if(music_enabled){
```
Comment (alegemaate) [inline BallBounceGame/Assets/SettingsGUI/SettingsScript.cs]: See comment above

File: BallBounceGame/Assets/SettingsGUI/SettingsScript.cs [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    public void ControlButtonPressed(){
+
+        control_state = !control_state;
+
+        PlayerPrefs.SetInt("Control", control_state ? 1 : 0);
+
+        ColorBlock new_cb = control_button.colors;
+        string new_text;
+
+        if(!control_state){
```
Comment (alegemaate) [inline BallBounceGame/Assets/SettingsGUI/SettingsScript.cs]: See comment above

File: BallBounceGame/Assets/SettingsGUI/SettingsScript.cs [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+        if(PlayerPrefs.GetInt("Music", 1)==0)
+            MusicButtonPressed();
+        
+        if(PlayerPrefs.GetInt("Sound", 1)==0)
+            SFXButtonPressed();
+        
+
+    }
+    public void ControlButtonPressed(){
```
Comment (alegemaate) [inline BallBounceGame/Assets/SettingsGUI/SettingsScript.cs]: Ctrl+K, Ctrl+D in visual studio

**Review by alegemaate — APPROVED** [review]

File: BallBounceGame/Assets/SettingsGUI/SettingsScript.cs (hunk 7 above)
Comment (danwardvs) [reply BallBounceGame/Assets/SettingsGUI/SettingsScript.cs]: Ctrl+Shift+I for linux there good sir

**Pull request merged**